modules/javafx.graphics/src/main/java/com/sun/glass/ui/monocle/MouseState.java

```diff
@@ -78,19 +78,26 @@ void releaseButton(int button) {
         buttonsPressed.removeInt(button);
     }
 
-    /** Returns the Glass window on which the coordinates of this state are located.
+    /**
+     * Returns the Glass window on which the coordinates of this state are located.
      * @param recalculateCache true if the cached value for the target window
      *                         should be recalculated; false if the cached
      *                         value should be used to determine the result
      *                         of this method.
+     * @param fallback if the original window is null, or if no window can
+     * be found, return the fallback window
      * @return the MonocleWindow at the top of the stack at the coordinates
-     * described by this state object.
+     * described by this state object, or the fallback window in case the
+     * current window is null or no window can be found for the supplied coordinates.
      */
-    MonocleWindow getWindow(boolean recalculateCache) {
-        if (window == null || recalculateCache) {
+    MonocleWindow getWindow(boolean recalculateCache, MonocleWindow fallback) {
+        if (recalculateCache) {
```

Comment:
Two questions:

1. It looks like `fallback` is effectively unused (always null). Did you add this for a future use case?
2. Unless I'm missing something, the only behavioral change is that it no longer executes the `if` block when the window is null if `recalculateCache` is false.

Reply:
true to both questions.
The reason for this change is that this is now doing the same as what is done in `monocle.TouchState` and `monocle.TouchInput`. 
TBH, I would prefer a different approach, where no fallback is passed, and the `if` block is only executed when `recalculateCache` is set to `true`. The only case where this `getWindow` method is invoked, is from `MouseInput` and it is only used to detect if the "old" window is different from the current window. Hence, it all seems a bit overkill to me, but for some reason, I thought keeping things consistent (mouseState versus TouchState) would make sense.
If you are ok with it, I can simply rewrite this without looking at the TouchState case?

Reply:
Either is fine with me. As long as this fixes your problem, go ahead with the fix. Or if you prefer, go ahead and rewrite it.

Reply:
I'll keep it like this for now, and I created a follow-up issue (JDK-8267175) for dealing with touch/mouse/key state/input handling in general in monocle.